elytron/src/test/java/org/wildfly/extension/elytron/CertificateAuthoritiesTestCase.java

```diff
@@ -175,6 +179,75 @@ public void testCreateAccount() throws Exception {
         }
     }
 
+    @Test
+    public void testCreateAccountWithCustomCA() throws Exception {
+        addKeyStore(ACCOUNTS_KEYSTORE_NAME);
+        addCertificateAuthorityWithoutStagingUrl();
+        addCertificateAuthorityAccountWithCustomCA("account1");
+        server = setupTestCreateAccount();
+        AcmeAccount acmeAccount = getAcmeAccount();
+        final String NEW_ACCT_LOCATION = "http://localhost:4001/acme/acct/384";
+        try {
+            assertNull(acmeAccount.getAccountUrl());
+            ModelNode operation = new ModelNode();
+            operation.get(ClientConstants.OP_ADDR).add("subsystem", "elytron").add("certificate-authority-account", CERTIFICATE_AUTHORITY_ACCOUNT_NAME);
+            operation.get(ClientConstants.OP).set(ElytronDescriptionConstants.CREATE_ACCOUNT);
+            operation.get(ElytronDescriptionConstants.AGREE_TO_TERMS_OF_SERVICE).set(true);
+            assertSuccess(services.executeOperation(operation));
+            assertEquals(NEW_ACCT_LOCATION, acmeAccount.getAccountUrl());
+        } finally {
+            removeCertificateAuthorityAccount();
+            removeCertificateAuthority();
+            removeKeyStore(ACCOUNTS_KEYSTORE_NAME);
+        }
+    }
+
+    @Test
+    public void testCreateAccountWithStagingAndCustomCA() throws Exception {
+        addKeyStore(ACCOUNTS_KEYSTORE_NAME);
+        addCertificateAuthorityWithStagingUrl();
+        addCertificateAuthorityAccountWithCustomCA("account1");
+        server = setupTestCreateAccount();
+        AcmeAccount acmeAccount = getAcmeAccount();
+        final String NEW_ACCT_LOCATION = "http://localhost:4001/acme/acct/384";
+        try {
+            assertNull(acmeAccount.getAccountUrl());
+            ModelNode operation = new ModelNode();
+            operation.get(ClientConstants.OP_ADDR).add("subsystem", "elytron").add("certificate-authority-account", CERTIFICATE_AUTHORITY_ACCOUNT_NAME);
+            operation.get(ClientConstants.OP).set(ElytronDescriptionConstants.CREATE_ACCOUNT);
+            operation.get(ElytronDescriptionConstants.AGREE_TO_TERMS_OF_SERVICE).set(true);
+            operation.get(ElytronDescriptionConstants.STAGING).set(true);
```

Comment:
@Skyllarr 
We should also check that CA can be added with both url and staging-url and CAA operation create-account won’t fail with url (not staging-url). I don't see this test in this PR ...

Reply:
This test has been added.